static/js/components/TopAppBar_test.js

```diff
@@ -33,37 +33,73 @@ describe("TopAppBar component", () => {
 
     it("has a button to collapse the menu", () => {
       assert.isOk(
-        shallow(<TopAppBar currentUser={user} />)
+        shallow(<TopAppBar currentUser={user} location={null} />)
           .find("button")
           .exists()
       )
     })
+
+    it("does not have a login/register on ecommerce bulk page", () => {
+      const location = { pathname: "/ecommerce/bulk/", hash: "", search: "" }
+      const wrapper = shallow(
+        <TopAppBar currentUser={user} location={location} />
+      )
+      assert.isNotOk(wrapper.find("UserMenu").exists())
+      assert.isNotOk(wrapper.find("MixedLink").exists())
+    })
+
+    it("does not have a login/register on ecommerce bulk receipt page", () => {
+      const location = { pathname: "/ecommerce/bulk/", hash: "", search: "" }
+      const wrapper = shallow(
+        <TopAppBar currentUser={user} location={location} />
+      )
+      assert.isNotOk(wrapper.find("UserMenu").exists())
+      assert.isNotOk(wrapper.find("MixedLink").exists())
+    })
   })
   describe("for logged in users", () => {
     const user = makeUser()
 
     it("has a UserMenu component", () => {
       assert.isOk(
-        shallow(<TopAppBar currentUser={user} />)
+        shallow(<TopAppBar currentUser={user} location={null} />)
           .find("UserMenu")
           .exists()
       )
     })
 
     it("does not have a button to collapse the menu", () => {
       assert.isNotOk(
-        shallow(<TopAppBar currentUser={user} />)
+        shallow(<TopAppBar currentUser={user} location={null} />)
           .find("button")
           .exists()
       )
     })
 
     it("does not have MixedLink's for login/registration", () => {
       assert.isNotOk(
-        shallow(<TopAppBar currentUser={user} />)
+        shallow(<TopAppBar currentUser={user} location={null} />)
           .find("MixedLink")
           .exists()
       )
     })
+
+    it("does not have a login/register on ecommerce bulk page", () => {
```

Comment:
2 same identical tests are in that file ?

Reply:
One set is for logged in users and the other is for anonymous users